### PR TITLE
Extend due date workaround

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -51,6 +51,9 @@ compose() {
 (
   cd "${projectRoot}"
 
+  # Create test output folder
+  mkdir -p test-output
+
```
Comment (johnwatson484) [inline scripts/test:56]: Ensures that directory is created with correct permissions

File: scripts/test [hunk 2]
```diff
@@ -61,11 +64,8 @@ compose() {
 
   # Run tests
   if [ "${debug}" = "true" ]; then
-    compose run --service-ports ${service} ${command}
+    compose run --service-ports --rm ${service} ${command}
   else
-    compose run ${service} ${command}
+    compose run --rm ${service} ${command}
```
Comment (johnwatson484) [inline scripts/test:69]: Automatically clean up migration containers, rather than them building up

File: scripts/test [hunk 3]
```diff
@@ -61,11 +64,8 @@ compose() {
...
   fi
-
-  # Clean up
-  compose down -v
```
Comment (johnwatson484) [inline scripts/test:70]: This adds no value

File: app/processing/siti-agri/handle-known-defects/correct-invalid-due-dates/invalid-due-dates.js [hunk 4]
```diff
@@ -25,4 +25,49 @@ module.exports = [{
 }, {
   invalid: '2023-03-05',
   correction: '2023-06-05'
+}, {
```
Comment (simonnineteen90) [inline app/processing/siti-agri/handle-known-defects/correct-invalid-due-dates/invalid-due-dates.js:28]: Is there a reason that we set our `invalid` dates to be on the 5th of each month `2022-08-05` but the table in the user story states first of the month.
image

**Review by simonnineteen90 — COMMENTED** [review]
What happens after 2024 when we reach the end of this list? Will we have to do this again or is there a longer term solution in the works?

File: app/processing/siti-agri/handle-known-defects/correct-invalid-due-dates/invalid-due-dates.js (hunk 4 above)
Comment (johnwatson484) [reply app/processing/siti-agri/handle-known-defects/correct-invalid-due-dates/invalid-due-dates.js:28]: The agreement start dates are the 1st, Siti Agri maps them to the 5th.   But the 5th of the wrong month.  You're right though it's not quite clear in the table what the ask is.  But 5th -> 5th is correct as per existing mapping

**Comment by johnwatson484** [comment]
> What happens after 2024 when we reach the end of this list? Will we have to do this again or is there a longer term solution in the works?

Hopefully Siti Agri will be updated by then.  But if not same as we've done now we've reached the end of this list.  I did put forward the option that we map everything forward three months, but agreed we'd stick to a fixed list.

**Review by simonnineteen90 — APPROVED** [review]

**Pull request merged**